docs/nightly/en/user-guide/continuous-aggregation/query.md

```diff
@@ -18,3 +18,86 @@ The query should have a `FROM` clause to identify the source table. As the join
 `GROUP BY` clause works as in a normal query. It groups the data by the specified columns. One special thing is the time window functions `hop()` and `tumble()` described in [Define Time Window](./define-time-window.md) part. They are used in the `GROUP BY` clause to define the time window for the aggregation. Other expressions in `GROUP BY` can be either literal, column or scalar expressions.
 
 Others things like `ORDER BY`, `LIMIT`, `OFFSET` are not supported in the continuous aggregation query.
+
+## Rewrite existing query to continuous aggregation query
+
+Some of simple existing aggregation queries can be directly used as continuous aggregation queries. For example, the example in [overview](./overview.md) can be used to query both in standard SQL and continuous aggregation query, since it's also a valid sql query without any flow-specific syntax or function:
+
+```sql
+SELECT
+    status,
+    count(client) AS total_logs,
+    min(size) as min_size,
+    max(size) as max_size,
+    avg(size) as avg_size,
+    sum(case when `size` > 550::double then 1::double else 0::double end) as high_size_count,
+    date_bin(INTERVAL '1 minutes', access_time) as time_window,
+FROM ngx_access_log
+GROUP BY
+    status,
+    time_window;
+```
+
+However, there are other types of query that cannot be directly used as continuous aggregation query. 
+For example, a query that need to compute percentile, because it would be unwise to repeatedly calculate the percentile for each time window. In this case, you can pre-aggregate the data into buckets of desired size, and then calculate the percentile in the sink table using standard SQL when needed. The original sql might be:
+```sql
+SELECT
+    status,
+    percentile_approx(size, 0.5) as median_size,
+    date_bin(INTERVAL '1 minutes', access_time) as time_window,
+FROM ngx_access_log
+GROUP BY
+    status,
+    time_window;
+```
+The above query will then be rewritten to first aggregate the data into buckets of size 10, and then calculate the percentile in the sink table.
+The flow query would be:
+```sql
+CREATE FLOW calc_ngx_distribution
+SINK TO ngx_distribution
+AS
+SELECT
+    status,
+    trunc(size, -1) as bucket,
+    count(client) AS total_logs,
+    date_bin(INTERVAL '1 minutes', access_time) as time_window,
+FROM ngx_access_log
+GROUP BY
+    status,
+    time_window,
+    bucket;
+```
+
+And then you can calculate the percentile in the sink table using standard SQL:
+```sql
+SELECT
+    outer.status,
+    outer.time_window,
+    outer.bucket,
+    SUM(case when in1.bucket <= outer.bucket then in1.total_logs else 0 end) * 100 / SUM(in1.total_logs) AS percentile
+FROM ngx_distribution AS outer
+JOIN ngx_distribution AS in1
+ON in1.status = outer.status 
+AND in1.time_window = outer.time_window
+GROUP BY
+    status,
+    time_window,
+    bucket
+ORDER BY status, time_window, bucket;
+```
```

Comment:
This seems to be a user-unfriendly way of syntax.
But if there is no other way, let it be.